simulator/func_sim/rf/rf.h

```diff
@@ -34,18 +34,20 @@ class RF
         return get_entry( num).value;
     }
 
-    void write( Register num, uint64 val)
+    void write( Register num, uint64 val, bool accumulating_instr = false)
     {
         if ( num.is_zero())
             return;
         if ( num.is_mips_hi_lo()) {
-            write( Register::mips_hi, val >> 32);
-            write( Register::mips_lo, val);
+            write( Register::mips_hi, val >> 32, accumulating_instr);
```

Comment:
It has a bug. Consider that HI contains `0x0`, and LO contains `~0x0` (all ones), and multiplication result is `1`. Your implementation will add `1` to LO and `0` to HI, making both registers equal to zero. The correct behavior is to handle carry to `HI` as well. I think you need to read HI and LO values here, combine them into one, do addition (or substraction), and write new values.